pkg/leader/leaderelector.go

```diff
@@ -0,0 +1,120 @@
+package leader
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+)
+
+const (
+	configMapName = "assisted-service-leader-election-helper"
+)
+
+type Config struct {
+	LeaseDuration time.Duration `envconfig:"LEADER_LEASE_DURATION" default:"15s"`
+	RetryInterval time.Duration `envconfig:"LEADER_RETRY_INTERVAL" default:"2s"`
+	RenewDeadline time.Duration `envconfig:"LEADER_RENEW_DEADLINE" default:"10s"`
+	Namespace     string        `envconfig:"NAMESPACE" default:"assisted-installer"`
+}
+
+//go:generate mockgen -source=leaderelector.go -package=leader -destination=mock_leader_elector.go
+
+type ElectorInterface interface {
```

Comment:
maybe split this interface so that monitors get only `IsLeader` and not `StartLeaderElection`
```
type Leader interface {
	IsLeader() bool
}
type ElectorInterface interface {
	Leader
	StartLeaderElection(ctx context.Context) error
}
```